Gems/Atom/Asset/Shader/Code/Source/Editor/ShaderVariantAssetBuilder.cpp

```diff
@@ -1140,10 +1141,137 @@ namespace AZ
                 }
             }
 
+            if (shaderVariantInfo.m_enableRegisterAnalysis && strcmp(creationContext.m_shaderPlatformInterface.GetAPIName().GetCStr(), "vulkan") == 0)
+            {
+                AZ::IO::FixedMaxPath projectBuildPath = AZ::Utils::GetExecutableDirectory();
+                projectBuildPath = projectBuildPath.RemoveFilename(); // profile
+                projectBuildPath = projectBuildPath.RemoveFilename(); // bin
+
+                AZStd::string spirvPath;
+                AZStd::string spirvFile = AZStd::string::format(
+                    "%s_vulkan_%u.spirv.bin", creationContext.m_shaderStemNamePrefix.c_str(), shaderVariantInfo.m_stableId);
+                AZ::StringFunc::Path::Join(creationContext.m_tempDirPath.c_str(), spirvFile.c_str(), spirvPath, true, true);
```

Comment:
This can use the operator/= or Append function of AZ::IO::Path
```suggestion
                AZ::IO::FixedMaxPath spirvPath(AZStd::string_view(creationContext.m_tempDirPath));
                spirvPath /= AZ::IO::FixexMaxPathString::format(
                    "%s_vulkan_%u.spirv.bin", creationContext.m_shaderStemNamePrefix.c_str(), shaderVariantInfo.m_stableId);
```